src/app/shared/shared.module.ts

```diff
@@ -4,22 +4,41 @@ import { EmailValidatorDirective } from './directives/email-validator.directive'
 import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
 import { ButtonModule } from './components/button/button.module';
 import { ButtonComponent } from './components/button/button.component';
+import { SearchModule } from './components/search/search.module';
+import { FormsModule } from '@angular/forms';
+import { DurationPipe } from './pipes/duration.pipe';
+import { CreationDatePipe } from './pipes/creation-date.pipe';
+import { StringJoinerPipe } from './pipes/string-joiner.pipe';
+import { TogglePasswordDirective } from './directives/toggle-password.directive';
+import { ControlErrorDirective } from './directives/control-error.directive';
 
 
 
 @NgModule({
   declarations: [
     EmailValidatorDirective,
+    DurationPipe,
+    CreationDatePipe,
+    StringJoinerPipe,
+    TogglePasswordDirective,
+    ControlErrorDirective,
```

Comment:
Since it is usual in shared module that `declarations` and `exports` lists are the same, we can add a variable to prevent code duplication.
```
const exportList = [...];
...
@NgModule({
  declarations: exportList,
  exports: exportList,
... 
```